cmake/CompilerFlags.cmake

```diff
@@ -99,14 +100,16 @@ ELSEIF ( CMAKE_COMPILER_IS_GNUCXX OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang"
       ADD_CXX_DEFINITIONS("-Wno-invalid-source-encoding")
     endif()
 
-    # ADDITIONAL DEBUG-MODE-SPECIFIC FLAGS
+    # ADDITIONAL GCC-SPECIFIC FLAGS
     if ( CMAKE_COMPILER_IS_GNUCXX ) # g++
       ADD_CXX_DEBUG_DEFINITIONS("-ffloat-store") # Improve debug run solution stability
       ADD_CXX_DEBUG_DEFINITIONS("-fsignaling-nans") # Disable optimizations that may have concealed NaN behavior
       ADD_CXX_DEBUG_DEFINITIONS("-D_GLIBCXX_DEBUG") # Standard container debug mode (bounds checking, ...)
+      ADD_CXX_RELEASE_DEFINITIONS("-finline-limit=2000") # More aggressive inlining
     endif()
 
   ADD_CXX_DEBUG_DEFINITIONS("-ggdb") # Produces debugging information specifically for gdb
+  ADD_CXX_RELEASE_DEFINITIONS("-Ofast -fno-stack-protector") # -Ofast (or -ffast-math) needed to auto-vectorize floating point loops
```

Comment:
@DeadParrot I know this will definitely improve the compiler's ability to auto vectorize, but this might need some thought and discussion. There will definitely be diffs due to this. `-Ofast` adds `-ffast-math` like you mention, which no longer makes floats (and doubles) IEEE compliant. I am unsure how I feel about this change personally at this moment.

@Myoldmopar fyi